src/libstore/content-address.hh

```diff
@@ -22,7 +22,7 @@ namespace nix {
  * Somewhat obscure, used by \ref Derivation derivations and
  * `builtins.toFile` currently.
  */
-struct TextHashMethod : std::monostate { };
+struct TextIngestionMethod : std::monostate { };
```

Comment:
Maybe there should be an explanation somewhere of exactly what the difference is between `TextIngestionMethod` and `FileIngestionMethod::Flat`. 

Here's the what I could come up with:

- they are hashed identically but tagged differently: `text:${algo}:${hash}` instead of `fixed:${algo}:${hash}`
- text-ingested store paths may not have self-references

There must be something else that I'm missing though, right?  It seems like it's just one extra bit on `FileIngestionMethod::Flat` and a no-self-references assertion.

Reply:
Yup that's correct. The emphasis for me is on the first one: ultimately `ContentAddressWithReferences` is supposed to represent every of content-addressed store path. `Flat` and `Text` may actually be the same thing, but so long as `text:` and `fixed:` are not, then they are different *by fiat*.

Reply:
Please mark this resolved.